src/Groups/matrices/form_group.jl

```diff
@@ -0,0 +1,654 @@
+
+export
+    invariant_alternating_forms,
+    invariant_bilinear_forms,
+    invariant_hermitian_forms,
+    invariant_quadratic_forms,
+    invariant_sesquilinear_forms,
+    invariant_symmetric_forms,
+    isometry_group,
+    preserved_quadratic_forms,
+    preserved_sesquilinear_forms
+
+########################################################################
+#
+# From group to form #TODO: there are different approaches. Which is the best?
+#
+########################################################################
+
+# TODO 1st approach: brute force calculation
+# Algorithm furnished by Thomas Breuer, Aachen University
+# extended to quadratic by Giovanni De Franceschi, TU Kaiserslautern
+# WARNING: big linear system !!
+
+# METHOD: given a group G, the condition gBg*=B is a linear condition on the coefficients of B,
+# hence we write down such system of dimension n^2 (n*(n-1)/2 for quadratic forms)
+
+"""
+    invariant_bilinear_forms(G::MatrixGroup)
+
+Return a generating set for the vector spaces of bilinear forms preserved by the group `G`.
+!!! warning "Note:"
+    At the moment, elements of the generating set are returned of type `mat_elem_type(G)`.
+"""
+function invariant_bilinear_forms(G::MatrixGroup{S,T}) where {S,T}
+   F = base_ring(G)
+   n = degree(G)
+   M = T[]
+   for mat in gens(G)
+      mmat = mat^-1
+      MM = zero_matrix(F,n^2,n^2)
+      for i in 1:n, j in 1:n, k in 1:n
+         MM[n*(i-1)+j,n*(k-1)+j] += mat[i,k]
+         MM[n*(i-1)+j,n*(i-1)+k] -= mmat[j,k]
+      end
+      push!(M,MM)
+   end
+
+   r,K = nullspace(block_matrix(length(M),1,M))
+   
+   return [matrix(F,n,n,[K[i,j] for i in 1:n^2]) for j in 1:r]
+end
+
+"""
+    invariant_sesquilinear_forms(G::MatrixGroup)
+
+Return a generating set for the vector spaces of sesquilinear non-bilinear forms preserved by the group `G`.
+It works only if `base_ring(G)` has even degree.
+!!! warning "Note:"
+    At the moment, elements of the generating set are returned of type `mat_elem_type(G)`.
+"""
+function invariant_sesquilinear_forms(G::MatrixGroup{S,T}) where {S,T}
+   F = base_ring(G)
+   @assert typeof(F)<:FinField "At the moment, only finite fields are considered"
+   @assert iseven(degree(F)) "Base ring has no even degree"
+   n = degree(G)
+   M = T[]
+   for mat in gens(G)
+      mmat = map(y->frobenius(y,div(degree(F),2)),(mat.elm)^-1)
+      MM = zero_matrix(F,n^2,n^2)
+      for i in 1:n, j in 1:n, k in 1:n
+         MM[n*(i-1)+j,n*(k-1)+j] += mat[i,k]
+         MM[n*(i-1)+j,n*(i-1)+k] -= mmat[j,k]
+      end
+      push!(M,MM)
+   end
+
+   r,K = nullspace(block_matrix(length(M),1,M))
+   
+   return [matrix(F,n,n,[K[i,j] for i in 1:n^2]) for j in 1:r]
+end
+
+"""
+    invariant_quadratic_forms(G::MatrixGroup)
+
+Return a generating set for the vector spaces of quadratic forms preserved by the group `G`.
+!!! warning "Note:"
+    At the moment, elements of the generating set are returned of type `mat_elem_type(G)`.
```

Comment:
By the way, I don't quite understand these notes: When you say "At the moment", it makes me think "... so there is a plan to do something else", and I wonder "but what"... Care to elaborate what you have in mind there?

Reply:
One may expect to have an output of type `SesquilinearForm` instead of `MatElem`. I don't know if in future this can change, i.e. one will change these functions in order to have an output of type `SesquilinearForm`, that's why I wrote "At the moment". For now, I can just write that the output has type `MatElem`.